pkg/deploy/api/validation/validation.go

```diff
@@ -32,6 +32,9 @@ func ValidateDeploymentConfigUpdate(newConfig *deployapi.DeploymentConfig, oldCo
 	allErrs := fielderrors.ValidationErrorList{}
 	allErrs = append(allErrs, validation.ValidateObjectMetaUpdate(&newConfig.ObjectMeta, &oldConfig.ObjectMeta).Prefix("metadata")...)
 	allErrs = append(allErrs, ValidateDeploymentConfig(newConfig)...)
+	if newConfig.LatestVersion < oldConfig.LatestVersion {
```

Comment:
It should always be >= 0, and although gaps resulting from increments of >1 shouldn't really be a problem functionally, I think it makes sense to keep it as contiguous as possible.